joomla/joomla-cms
# [4.0] System Panel: Add badges to items when necessary

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/23032

### Summary of Changes
As title says.
### Testing Instructions
1. Checkout some items.
2. Modify your php.ini: for example set `file_uploads = Off` to get a Warning
3. Modify `administrator/manifests/files/joomla.xml`
 to
`<version>4.0.0-alpha4</version>`
and `libraries/src/Version.php`
to `const EXTRA_VERSION = 'alpha4-dev';`
in order to get a database fix issue
4. Add a module or plugin to be discovered.
5. Install an older language pack in order to get a warning for an update.
[fr-FR_joomla_lang_full_3.9.0v1.zip](https://github.com/joomla/joomla-cms/files/2570924/fr-FR_joomla_lang_full_3.9.0v1.zip)

Note: Although we should get also a pill for a joomla update with this patch, I did not find a way to test that.

EDIT: see https://github.com/joomla/joomla-cms/pull/23047#issuecomment-437835801 to test joomlaupdate

### After patch

<img width="1364" alt="screen shot 2018-11-12 at 07 17 52" src="https://user-images.githubusercontent.com/869724/48332494-4390bb00-e654-11e8-9958-bbbc098901c1.png">
